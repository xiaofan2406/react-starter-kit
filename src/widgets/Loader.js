/**
 * @see https://github.com/yuanyan/halogen/blob/master/src/RingLoader.js
 */
import React from 'react';
import PropTypes from 'prop-types';
import { css } from '@emotion/core';

const Loader = ({ size, color }) => (
  <div
    css={css`
      display: inline-block;
    `}
  >
    <div
      css={css`
        width: ${size}px;
        height: ${size}px;
        position: relative;

        & div {
          width: ${size}px;
          height: ${size}px;
          border-style: solid;
          border-width: ${size / 10}px;
          border-color: ${color};
          opacity: 0.4;
          border-radius: 100%;
          position: absolute;
          top: 0;
          left: 0;
          perspective: 800px;
          animation-fill-mode: forwards;
        }
      `}
    >
      <div
        css={css`
          animation: loaderRigthRotate 2s 0s infinite linear;
        `}
      />
      <div
        css={css`
          animation: loaderLeftRotate 2s 0s infinite linear;
        `}
      />
    </div>
  </div>
);

Loader.propTypes = {
  color: PropTypes.string,
  size: PropTypes.number,
};

Loader.defaultProps = {
  color: '#00bcd4',
  size: 60,
};

export default Loader;
