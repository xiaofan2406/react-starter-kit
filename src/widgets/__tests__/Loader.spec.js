import React from 'react';
import { render } from '@testing-library/react';
import { expect, test } from '@jest/globals';
import Loader from '../Loader';

test('Loader matches snapshot', () => {
  const { container } = render(<Loader />);

  expect(container.firstChild).toMatchSnapshot();
});
