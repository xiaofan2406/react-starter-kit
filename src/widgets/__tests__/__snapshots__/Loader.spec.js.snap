// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader matches snapshot 1`] = `
.emotion-3 {
  display: inline-block;
}

.emotion-2 {
  width: 60px;
  height: 60px;
  position: relative;
}

.emotion-2 div {
  width: 60px;
  height: 60px;
  border-style: solid;
  border-width: 6px;
  border-color: #00bcd4;
  opacity: 0.4;
  border-radius: 100%;
  position: absolute;
  top: 0;
  left: 0;
  -webkit-perspective: 800px;
  -moz-perspective: 800px;
  -ms-perspective: 800px;
  perspective: 800px;
  -webkit-animation-fill-mode: forwards;
  animation-fill-mode: forwards;
}

.emotion-0 {
  -webkit-animation: loaderRigthRotate 2s 0s infinite linear;
  animation: loaderRigthRotate 2s 0s infinite linear;
}

.emotion-1 {
  -webkit-animation: loaderLeftRotate 2s 0s infinite linear;
  animation: loaderLeftRotate 2s 0s infinite linear;
}

<div
  class="emotion-3"
>
  <div
    class="emotion-2"
  >
    <div
      class="emotion-0"
    />
    <div
      class="emotion-1"
    />
  </div>
</div>
`;
